Add Necker Cube study

sebastiansauer opened the pull request:
This study investigates attentional processes using a well-known bistabile image paradimga. Precisely, participants watch the Necker cube, a simple wire frame cube. The necker cube can be perceived in one of two perspectives. After a couple of seconds, the perspectives "flips" from one perspective to the other. This change cannot (readily) suspended by will. The flip span can be associated with the temporal expansion of the working memory (Sauer et al., 2012).


Literature
Sauer, S., Lemke, J., Wittmann, M., Kohls, N., Mochty, U., & Walach, H. (2012). How long is now for mindfulness meditators? Personality and Individual Differences, 52(6), 750–754. https://doi.org/10.1016/j.paid.2011.12.026

Access:
https://s3.amazonaws.com/academia.edu.documents/44855192/How_long_is_now_for_mindfulness_meditato20160418-529-obhjyj.pdf?AWSAccessKeyId=AKIAIWOWYYGZ2Y53UL3A&Expires=1549041333&Signature=EuubJvXVF3j5vCF6E70H7EBfqB8%3D&response-content-disposition=inline%3B%20filename%3DHow_long_is_now_for_mindfulness_meditato.pdf

FelixHenninger commented:
Hej @sebastiansauer , 

wow, this is awesome, so many thanks for contributing back! I'd be happy to merge this into the repository as-is (please don't worry about the failing test, that's on me), but if you can spare another moment, here are three things we could take a look at (I can do these, too, if you'd prefer):

* Just recently, we added the ability to include files in the experiment itself, which makes it easier to run studies offline, and independently of changing URLs elsewhere -- maybe we could think about bundling your stimulus, too? Right now, the image on Wikipedia is CC-BY-SA-licensed, so including that would affect the study license. I would be happy to draw you a new stimulus, if you'd like?
* Would you like to put your description in the study metadata?
* On the `Necker show` screen, the footer is jumping a little -- I think you could remove that by removing the `<div class="container">` element that surrounds the `<main>` stimulus?

Ok, so much for now -- again, this is super-cool, and I would love to add this to our study collection (I really enjoyed your paper, too!). Let me know how you'd like to proceed, and we'll go from there.

All the very best,

-Felix

sebastiansauer commented:
Hi @FelixHenninger,

Thanks for your note! That sounds all completely reasonable, I'll be happy to fix the three points you mentioned. I hope to get that going in the next days. Would it be the easiest if I post at PR again?

Keep up the great work, I love lab.js, and I have already planned to teach it to my (undergrad psych) students in the summer term.

Are there any more examples outthere, which you are aware of? I think examples are of great help for the community. Well if not, I hope to convince my students to contribute!

Thanks again
Sebastian

FelixHenninger commented:
Hi @sebastiansauer,

thanks a lot for your comments! As I hope I've made clear, I really appreciate your addition -- the suggestions are just that :-) . I think that you can just update your repository, and the PR will get updated automatically, but I'm not entirely sure. Let me know if I can help you, I'd be glad to!

It's awesome to hear that you're planning to use `lab.js` in teaching! And yes, we are working on a set of examples (in fact, we got a grant from the FernUni and Wikimedia to put together a demo library). You can [see a couple of those tasks in action](http://psychexphagen.fernuni-hagen.de/Experimente/) via their homepage, if you have a suggestions, do let me know!

Ok, so much for now; thank _you_, you made my day!

-Felix

sebastiansauer commented:
Hi @FelixHenninger,

the examples are cool! I am happy to contribute event it's just one more example. 

Thanks!
Sebastian

sebastiansauer commented:
Hi Felix,

I just did three changes:

1) I've added a self-painted picture to the test, replacing the Wiki images, and subject to the licence of lab.js
2) I've updated the meta data of the study
3) I've excluded my name/university in order to privide a general template
4) I've taken out the div-container.

I'll push all that. Let me know if I've to do something with the PR. 

Keep up the cool stuff! 👍

sebastiansauer commented:
Hi Felix, 

let me know if there's anything I can do from my end at the moment in this regard.

FelixHenninger commented:
Hej Sebastian,

thanks for the reminder, and sorry for this taking so long! I've been giving workshops and finishing features on a deadline over the past weeks, so things have been very busy around here.

But regarding your study, thanks again not only for sharing it, but also for revising it so thoroughly! As before, it looks phantastically awesome. A few more quick notes, in case you have some thoughts:

* In the latest version, there's still some jumpiness in the second block -- I've tried to reduce it in the [attached file](https://github.com/FelixHenninger/lab.js/files/2958095/attention-span-study-using-the-necker-cube-bistabile-image-2019-03-12--18_28.study.json.zip), and added a consistent footer during the trial, does that make sense to you? In my feeling, it makes sense to center the fixation cross relative to the stimulus, but I have no idea what makes sense substantively -- no doubt you know more here.
* The access link above isn't working for me (I could swear that it did, though, because I read your paper when you started this PR!). My guess is that academia.edu changed something -- I guess we might have to link to PAID, after all? 
* You write that you've removed your contact info, but it's still visible on the last page, is that intentional?
* I saw that you've included loads of source material, which is phantastic, but it got me thinking, whether the `lab.js` repository is the right place to put this entire project -- I was wondering whether it might be better archived, on, say the [Open Science Framework](http://osf.io/), possibly alongside the data collected using the study. I'd be happy to add a link from within the repository to it and other similar studies, if this were an option for you? Either way, I'm happy to include your example.

Ok, so much for now -- again, thanks for revising this, and apologies for the long radio silence! Best,

-Felix

sebastiansauer commented:
Hi Felix,

Great to hearing from you! Thanks for your helpful comments; they all make sense. Let me check that out, and I'll check-in the update then.

BTW: No worries about later than usual response time, I know that feeling when deadlines trying to slap on the fingers :-)

sebastiansauer commented:
Hi Felix,

Sorry for the delay, I was burried under a surge of term paper grading. Anyway, that's what I did, referring to your comments:

1) I used your version with less "bumbiness" in the footer etc.
2) I changed the citation to the published paper
3) I removed my contact details (ie, anonymized them) on the last page
4) I uploaded the images/materials to a osf repo: https://osf.io/4y2jp/ (CC0). I'm unsure however how to delete from lab.js. Do I just have to delete the files before pushing the pull request? If otherwise, feel free to delete; I've added the URL to the osf repo in the study description.

I've pushed my repo. Let me know if I need to anything else. As a backup, the study file (as zip) I just pushed is [here](https://github.com/FelixHenninger/lab.js/files/3022399/attention-span-study-using-the-necker-cube-bistabile-image-.template.-2019-03-29--12_19.study.json.zip)
